include/avif/internal.h

```diff
@@ -126,11 +126,11 @@ void avifImageSetDefaults(avifImage * image);
 // Copies all fields that do not need to be freed/allocated from srcImage to dstImage.
 void avifImageCopyNoAlloc(avifImage * dstImage, const avifImage * srcImage);
 
-// Copies the samples from srcImage to dstImage. dstImage must be allocated.
+// Copies the planes from srcImage to dstImage. dstImage must be allocated.
 // srcImage and dstImage must have the same width, height, and depth.
 // If the AVIF_PLANES_YUV bit is set in planes, then srcImage and dstImage must have the same yuvFormat and yuvRange.
 // Ignores the gainMap field (which exists only if AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP is defined).
-void avifImageCopySamples(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes);
+void avifImageCopyPlanes(avifImage * dstImage, const avifImage * srcImage, avifPlanesFlags planes);
```

Comment:
Why would you like to export this function? I guess `avifImageCopy()` does not fulfill your use case because it does not copy samples to a view?

I am fine with returning `avifResult` even if it stays internal.

Reply:
In tests/gtest/aviftest_helpers.cc, the second `MergeGrid()` function calls `avifImageCopySamples()` and uses the `AVIF_CHECKERR` and `AVIF_CHECKRES` macros. This is why the aviftest_helpers library needs to depend on the avif_internal library, which contributes to the bug that some tests are linked with both the avif and the avif_internal libraries.

So exporting `avifImageCopySamples()` as a public function will allow aviftest_helpers to depend on the avif library, which may help fix the dual avif/avif_internal linking bug. A variant of this is to convert `MergeGrid()` to a public function.

But this is not the only way to fix the dual avif/avif_internal linking bug. So let's see how Vincent is planning to fix it.

Reply:
> So let's see how Vincent is planning to fix it.

See https://github.com/AOMediaCodec/libavif/pull/1819.